core/conversion/converters/impl/plugins/interpolate_plugin.cpp

```diff
@@ -0,0 +1,244 @@
+        input_archive.read("size", value);
+        size = value.toIntVector();
+    }
+    {
+        torch::IValue value;
+        input_archive.read("mode", value);
+        mode = value.toStringRef();
+    }
+    {
+        torch::IValue value;
+        input_archive.read("align_corners", value);
+        align_corners = value.toBool();
+    }
+}
+
+std::vector<int64_t> InterpolatePlugin::getInputShape() {
+    return in_shape;
+}
+
+std::vector<int64_t> InterpolatePlugin::getOutputShape() {
+    return out_shape;
+}
+
+std::vector<int64_t> InterpolatePlugin::getOutputSize() {
+    return size;
+}
+
+int InterpolatePlugin::getNbOutputs() const {
+    return 1;
+}
+
+const char* InterpolatePlugin::getPluginType() const {
+    return "Interpolate";
+}
+
+const char* InterpolatePlugin::getPluginVersion() const{
+    return "1";
+}
+
+const char* InterpolatePlugin::getPluginNamespace() const {
+    return "";
+}
+
+
+nvinfer1::IPluginV2DynamicExt* InterpolatePlugin::clone() const {
+    return new InterpolatePlugin(in_shape, out_shape, size, mode, align_corners);
+}
+
+nvinfer1::DimsExprs InterpolatePlugin::getOutputDimensions(int outputIndex, const nvinfer1::DimsExprs *inputs, int nbInputs, nvinfer1::IExprBuilder &exprBuilder) {
+   nvinfer1::DimsExprs output(inputs[0]);
+
+   for (unsigned int i = 0; i < out_shape.size(); i++) {
+       output.d[i] = exprBuilder.constant(out_shape[i]);
+   }
+
+   return output;
+}
+
+nvinfer1::DataType InterpolatePlugin::getOutputDataType(int index, const nvinfer1::DataType* inputTypes, int nbInputs) const {
+    return DataType::kFLOAT;
+}
+
+int InterpolatePlugin::initialize() {
+    tensor_options = tensor_options.device(c10::kCUDA);
+
+    // c10::kFloat = FLOAT32
+    tensor_options = tensor_options.dtype(c10::kFloat);
+
+    return 0;
+}
+
+
+void InterpolatePlugin::serialize(void* buffer) const {
+    std::string data = serializeToString();
+    size_t size = getSerializationSize();
+
+    data.copy((char*) buffer, size);
+}
+
+std::string InterpolatePlugin::serializeToString() const {
+    torch::serialize::OutputArchive output_archive;
+
+    output_archive.write("in_shape", torch::IValue(in_shape));
+    output_archive.write("out_shape", torch::IValue(out_shape));
+    output_archive.write("size", torch::IValue(size));
+    output_archive.write("mode", torch::IValue(mode));
+    output_archive.write("align_corners", torch::IValue(align_corners));
+
+    std::ostringstream data_str;
+    output_archive.save_to(data_str);
+
+    return data_str.str();
+}
+
+size_t InterpolatePlugin::getSerializationSize() const {
+    return serializeToString().size();
+}
+
+bool InterpolatePlugin::supportsFormatCombination(int pos, const nvinfer1::PluginTensorDesc* inOut, int nbInputs, int nbOutputs) {
+    if (inOut->format != nvinfer1::TensorFormat::kLINEAR) {
+        return false;
+    } 
+
+    if (inOut->type == DataType::kINT32 || inOut->type == DataType::kINT8) {
+        return false;
+    }
+

```

Comment:
Did you test with `--fp16` in either benchmark or `trtorchc`? I'm hoping it would just fall back on FP32